src/R/Editor/Impl/Completion/Providers/FilesCompletionProvider.cs

```diff
@@ -50,14 +50,18 @@ enum Mode {
             _forceR = forceR;
 
             _enteredDirectory = ExtractDirectory(directoryCandidate);
-            if (_enteredDirectory.Length == 0 || _enteredDirectory.StartsWithOrdinal(".")) {
+            _task = GetRootDirectoryAsync(_enteredDirectory);
+        }
+
+        private Task<string> GetRootDirectoryAsync(string userProvidedDirectory) {
```

Comment:
Why you don't want to make this method async? You don't have to use `Task.Run` in that case.